### ISSUE TITLE
App instantly marks session as expired after logging in

### ISSUE THREAD

**Issue by xefyrdev** [xefyrdev/discord-bio-links#13]
![App instantly marks session as expired after logging in](https://github.com/xefyrdev/discord-bio-links/assets/139470244/af95f373-4a16-455a-8af6-095ac7baa431)

**Comment by DarshanDixit05** [comment]
Hi @xefyrdev , I want to work on this issue. Can you please assign it to me?

**Comment by xefyrdev** [comment]
Hi, you've been assigned to the issue. On my side, I kept working to figure out what causes this bug but I've been unable to reproduce it on my machine. The console.log() statements I've added don't really help as everything is what I expect it to be.

**Comment by xefyrdev** [comment]
This is really weird.

```ts
// /code/server/controllers/users/lookup.ts

const internalUser = new InternalUser(req.headers.id);
const user: User = await internalUser.get();

const reply = { user };
res.json(reply);
console.log(`Replied to GET @ /api/users with`, reply);
```

The server logs me this:
<img width="960" alt="what the server replied with" src="https://github.com/xefyrdev/discord-bio-links/assets/139470244/d15d6486-3d60-40c8-8fde-8a87fc6f5c29">

Yet,
```js
// /public/js/helpers/request.js
.then(function (data) {
   console.log(`Fetched (${method} @ ${url}):`, data);
   resolve(data);
})
```

logs me this:
<img width="960" alt="fetchresult" src="https://github.com/xefyrdev/discord-bio-links/assets/139470244/76b20ba4-7e48-4e35-abe0-ebaf76cbafd0">

Since I am not able to reproduce this locally, I'm guessing it has to do with render (where it is deployed) or how I configured it.

**Comment by xefyrdev** [comment]
Making a Postman request to the API works so the issue definitely is in how I handle the server response.

![image](https://github.com/xefyrdev/discord-bio-links/assets/139470244/11002f86-89fe-4bcf-a102-a5eb3f87291f)

**Comment by DarshanDixit05** [comment]
Yes , we need to correct the handler functions. Currently im setting this locally.

**Comment by xefyrdev** [comment]
I resolved the issue. I assumed that if `Content-Length` was 0 or null then the body was empty and resolved the request promise with an empty object. Except that is not true. The Content-Length header was null even though the response had a body. (Maybe it's set automatically locally but not on render.com for some reason).

But all of my code relied on that response not being empty so they could find the current user and if they didn't find it, then it'd be because the Authorization token was incorrect (and thus the session expired).

I fixed this by always resolving the actual response body and resolved by the raw response if the request failed. (See e585ac0).
